src/txn/txn_recover.c

```diff
@@ -522,6 +532,15 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
 done:	FLD_SET(conn->log_flags, WT_CONN_LOG_RECOVER_DONE);
 err:	WT_TRET(__recovery_free(&r));
 	__wt_free(session, config);
+
+	/*
+	 * Destroy the eviction threads that were started in support of
+	 * recovery.  They will be restarted once the lookaside table is
+	 * created.
+	 */
+	if (eviction_started)
+		WT_TRET(__wt_evict_destroy(session));
+
```

Comment:
@michaelcahill, I'd suggest moving these calls outside of this function, wrapping `__wt_txn_recover`, if that's possible. Looking over the code, I don't see why it wouldn't be possible, but I could easily be missing something.